### DISCUSSION TITLE
[Proposal] Allow `default` keyword to define empty implementation of `void`-returning methods

### DISCUSSION THREAD

**Discussion by vadimkantorov** [dotnet/csharplang#7656] [Language Ideas]
E.g. `void Foo() {}` and `int Foo() => default;` are both currently possible but `void Foo() => default;` is currently not allowed. I propose to allow this syntax too, mainly for consistency

I'm not sure where to put this kind of suggestion. Please move it to issues or to another category if needed!

**Comment by iam3yal** [comment]
But then here `void Foo() {}` you're not returning anything whereas here `void Foo() => default;` you do so it's not really about consistency because the equivalent is this `void Foo() => ;` but either way I think that the former is more readable.

**Reply by vadimkantorov** [reply]
One could imagine "values" of the void type (similar to `Unit` type in F#) and `default` standing as a token for that `void` value in this method definition context. But I'm not extremely well-versed in the current design choices and philosophy of C# regarding the `void` type, so going to just float this minor proposal around.

**Reply by iam3yal** [reply]
@vadimkantorov Yes and there are some discussions about it. :)
